### ISSUE TITLE
mouse not working in Avast products on windows 10 

### ISSUE THREAD

**Issue by Fefo60** [Ylianst/MeshCentral#4163]
The mouse does not work on Avast Business Security windows
if I open the Windows Avast Business Security settings window and try to operate the mouse, the mouse does not respond in that window

**Server Software :**
 - OS: Windows 10pro
 - Version: 1.0.39

**Remote Device (please complete the following information):**
 - OS:  Windows 10pro
 - Version:  21H1
 - Current Core: Mar 25 2022, 2265086444

**Comment by Ylianst** [comment]
This is not a surprise and I am not sure we can do anything about it. These apps want to make sure that the local user is operating the mouse, not a remote user so this is by design. Not sure if there is anything we can do about this.

**Comment by PathfinderNetworks** [comment]
This is a known issue.  I also use Avast (Avast CloudCare) for all of my customers and have been having this same trouble for a few years now.  It's something to do with how Avast is doing security when you open the Avast client.  It blocks input from Mesh. But not other remote tools (like their own Premium Remote, Splashtop, etc).  If you send a Ctrl-Alt-Del it will accept that. But as soon as you click the Avast interface you'll lose input again.  
This is true of all Windows devices, not just Windows 10.  
It is pretty frustrating, but I haven't found a work around other than using a different remote tool when I need to access the Avast client remotely.

**Comment by PathfinderNetworks** [comment]
I wish we could figure out what the difference is in those other remote tools (like Remote Utilities, Splashtop, Premium Remote, etc) that will work with Avast for remote input while MeshCentral does not.  I'm certain it's a difference in how the input is being handled, but no idea what that difference would be.

**Comment by silversword411** [comment]
Kaspersky can do the same thing. I think it's custom programming, watching for .exe running, and having a special overlay on the av windows (obviously) they're trying to stop remote access from being able to turn off av controls

**Comment by Ylianst** [comment]
Bryan was supposed to work working on DirectX capture of the screen in the MeshAgent for a long time now, but it never got done. I don't think that impacts mouse/keyboard input, but yes, there are certainly differences between the different remote control software.

**Comment by veitw** [comment]
Not a Windows user, but I should not wonder if such security solutions decide on the basis of the program the input originates from. This could be simply whitelisted, but it could also be checking the code signature of the program the input originates from, and deciding on whether it is credible or not. This could be a list of software vendors to allow remote control from, but maybe it does suffice to sign the meshagent using a self-signed CA and adding this CA to Windows' CA trust?

**Comment by krayon007** [comment]
Yes, the directx APIs I was working on, is only for display capture, it didn't include input injection, so would still rely on SendInput. Let me see if I can see if there are other input APIs we can use.

**Comment by veitw** [comment]
Hi @PathfinderNetworks,

> I wish we could figure out what the difference is in those other remote tools (like Remote Utilities, Splashtop, Premium Remote, etc) that will work with Avast for remote input while MeshCentral does not.

did you have success with open source such as TightVNC (maybe tunneled via MeshCentral, or in reverse connection mode)?

I suppose this will help to find a way to make this work for MC, too.

**Comment by rraallvv** [comment]
Does anyone know if this issue was resolved, or at least what the feature in Avast that’s causing the problem is called?

**Comment by PathfinderNetworks** [comment]
> Does anyone know if this issue was resolved, or at least what the feature in Avast that’s causing the problem is called?

I still use Avast CloudCare (which uses the Avast Business Premium client/engine).  I haven't had this issue for a few years now.

**Comment by Fauzdar1** [comment]
I use DeskFlow and am not aware of MeshCentral. 

But, I landed to this issue from Google and I've found out a workaround. 

You can disable "Protection from Input Manipulation" in Avast's settings (Settings -> General -> Troubleshooting) to prevent it from blocking the inputs. 

![Image](https://github.com/user-attachments/assets/19253c41-181c-4e64-b913-0d2ea4a14e1c)

PS: This can be dangerous, and you should only do it if you know what you're doing.

As I use Deskflow to operate my local PC from my Mac using KVM so I disabled it. I interact with the system tray for my app, and it gets frustrating if the cursor gets on the Avast's icon and it blocks the whole KVM.
